Fix minimongo nested array sorting

sebakerckhof opened the pull request:
Fixes https://github.com/meteor/meteor/issues/10192 + added extra tests

sebakerckhof commented on packages/minimongo/sorter.js:
```diff
@@ -140,9 +140,9 @@ export default class Sorter {
+      // pretend we found one undefined value.
       if (!branches.length) {
-        branches = [{value: null}];
+        branches = [{value: undefined}];
```
null is sorted higher than undefined by mongo. But an empty array should be considered the same as undefined (ref: https://docs.mongodb.com/manual/reference/bson-type-comparison-order/#arrays)

sebakerckhof commented on packages/minimongo/sorter.js:
```diff
@@ -280,12 +280,6 @@ export default class Sorter {
       }
     });
 
-    // This could happen if our key filter somehow filters out all the keys even
```
Motivation: this was no longer necessary after https://github.com/meteor/meteor/pull/10214 since there's no longer a key filter.

mitar commented:
@sebakerckhof Are you interested in fixing more general list of Minimongo issues? I have [for this package](https://github.com/peerlibrary/meteor-reactive-mongo) tried to run all Minimongo tests on the MongoDB server-side and found quite some differences. I think Meteor should run all of client-side Minomongo query-related tests on server as well, to make sure this is compatible. (Ideally, we would run also standard MongoDB query-related tests on Minimongo, but that is much harder.)

sebakerckhof commented:
@mitar I do have a pretty good understanding of the minimongo code by now, so I could maybe identify some low hanging fruit. The most time-consuming part is usually figuring out the MongoDB logic, since often times the documentation is seriously lacking and you have to find out the MongoDB logic with trial & error.
But note that quite some parts of minimongo are known to be approximations of MongoDB behavior to reduce the code size and complexity (or sometimes it's just impossible to achieve in javascript), so don't expect to ever fully be able to pass all tests on the server.

Is it just a matter of changing `client` to `server` here: https://github.com/meteor/meteor/blob/devel/packages/minimongo/package.js#L46 ? Or how did you run those tests?

mitar commented:
Yes, I move tests to server, but then I have to patch few of them. See here for details: https://github.com/peerlibrary/meteor-reactive-mongo/blob/master/tests_prepare.js

I think what we should do is split tests into client-only, both, server-only. Then that patching would not be necessary. I think once you move to the server you quickly realize what should be where and what are differences. The [list I know of those which are probably things which should be fixed in Minimongo is in this section](https://github.com/peerlibrary/meteor-reactive-mongo/blob/master/tests_prepare.js#L89).

sebakerckhof commented:
@mitar : A lot of the failures are because of: https://github.com/meteor/meteor/issues/10023 ( https://docs.mongodb.com/manual/reference/limits/#Restrictions-on-Field-Names ).

In true mongodb fashion the behavior is weird and undocumented. But I've figured most out of it by trail and error now.
I've created a branch where I already updated the tests and added some new ones: https://github.com/sebakerckhof/meteor/tree/fix/10023.

However, fixing that is not going to be so easy.
